src/workflows-app/utils/method-common.js

```diff
@@ -96,3 +100,39 @@ export const getFilteredRuns = (filterOption, runsToFilter, errorStates) => {
     return false;
   });
 };
+
+export const getFilteredRunSets = (filterOption, runSetsToFilter, userId, errorStates) => {
+  return runSetsToFilter.filter((runSet) => {
+    switch (filterOption) {
+      case FilterOptions.Failed:
+        if (errorStates.includes(runSet.state)) {
+          return true;
+        }
+        break;
+      case FilterOptions.Succeeded:
+        if (runSet.state === 'COMPLETE') {
+          return true;
+        }
+        break;
+      case FilterOptions.OwnSubmissions:
+        if (runSet.user_id === userId) {
+          return true;
+        }
+        break;
+      default:
+        return true;
+    }
+    return false;
+  });
+};
+
+export const getSortableRunSets = (runSets, samId) => {
+  // samId must be a string representing a base16 integer
+  return _.map((rs) => _.merge({ submitter_priority: samId === rs.user_id ? -1 : parseInt(samId, 16) }, _.cloneDeep(rs)), runSets);
+};
+
+export const samIdToAnonymousName = (samId) => {
```

Comment:
Nice

Reply:
It would be nice to add unit tests for this method.

Reply:
+1

Reply:
good call, will do